fault-proof/tests/e2e.rs

```diff
@@ -941,4 +941,27 @@ mod e2e {
 
         Ok(())
     }
+
+    // Tests that the proposer fails fast when the contract's starting L2 block number is
+    // misconfigured to a future value (e.g., 1M blocks ahead of actual finalized L2 block).
+    // This prevents the proposer from running indefinitely without creating games.
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_proposer_rejects_future_starting_block() -> Result<()> {
+        let env = TestEnvironment::setup_with_starting_block_offset(1_000_000).await?;
```

Comment:
Let’s pick an offset value that determines exactly (from) where the rejection should happen, in this case, probably 1.

Reply:
[e266dd6](https://github.com/succinctlabs/op-succinct/pull/704/commits/e266dd67364a62f146c9ce4cbe505ce92c787e6c)